src/bson.rs

```diff
@@ -430,3 +466,121 @@ impl Bson {
         Bson::Document(values)
     }
 }
+
+/// Error for representing failure of getting BSON value
+#[derive(Debug)]
+pub struct GetBsonValueError;
+
+impl fmt::Display for GetBsonValueError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("GetBsonValueError")
+    }
+}
+
+impl Error for GetBsonValueError {
+    fn description(&self) -> &str {
+        "failed to get bson value"
+    }
+}
+
+/// Type alias of `Result<T, GetBsonValueError>`
+pub type GetBsonValueResult<T> = Result<T, GetBsonValueError>;
+
+/// Value helpers
+impl Bson {
+    /// If `Bson` is `FloatingPoint`, return its value. Returns `None` otherwise
+    pub fn as_f64(&self) -> GetBsonValueResult<f64> {
```

Comment:
in the interest of predictability and consistency with From/Into and TryFrom/TryInto, perhaps we should have TryAs?

```rust
pub trait TryAs<T> {
    type Err;
    fn try_as(&self) -> Result<T, Self::Err>;
}

bson = val.from();
into_val: ObjectId = bson.try_into()?;
into_val: &ObjectId = bson.try_as()?;
```